File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string str="";
        vector<string> ans;
        for(int i=0 ; i<s.size() ; i++) {
            if(s[i] == ' ') {
                if(str.size() > 0) {
                    ans.emplace_back(str);
                    str = "";
                }
                else continue;
            }
            else str += s[i];
        }
        if(str.size() > 0) {
            ans.emplace_back(str);
        }
        str = "";
        reverse(begin(ans), end(ans));
        for(int i=0 ; i<ans.size() ; i++) {
            str += ans[i];
            if(i != ans.size()-1) str += " ";
        }
        return str;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let arr = s.split(' ');
    arr.reverse();
    let res = '';
    arr.forEach((item) => {
        if (item.length !== 0) {
            res += item;
            res += ' ';
        }
    })
    return res.substr(0, res.length-1);
};